pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/internal/LanguageLevelChecker.java

```diff
@@ -221,6 +242,11 @@ private enum Keywords implements LanguageFeature {
          */
         PERMITS_AS_A_TYPE_NAME(15, "permits"),
 
+        /**
+         * ContextualKeyword since Java 19 Preview.
+         */
+        WHEN_AS_A_TYPE_NAME(19, "when"),
+
```

Comment:
I'm not sure, if this is correct... You *can* use `when` as a class name: `public class when {}` compiles fine with java 19 preview. Having this in the "Keywords" enum, we would prevent this, right? In that case, we probably should remove it...

Reply:
If we can use when as a type name then I suppose you're right, the only thing we need to add in this enum is the `CASE_REFINEMENT` above

Reply:
I'll remove it then